src/Augmint.ts

```diff
@@ -205,13 +200,128 @@ export class Augmint {
                 throw new Error("legacy contracts length mismatch!");
             }
         }
-        const options: ILoanManagerOptions = {
-            token: this.token, // FIXME: This should come from LoanManager contract's augmintToken property
-            rates: this.rates, // FIXME: This should come from LoanManager contract's rates property
-            ethereumConnection: this.ethereumConnection
-        };
         return legacyContracts.map(
-            (contract: DeployedContract<LoanManagerInstance>) => new LoanManager(contract.connect(this.web3), options)
+            (contract: DeployedContract<LoanManagerInstance>) =>
+                new LoanManager(contract.connect(this.web3), this.ethereumConnection)
         );
     }
+
+    public async getLoansForAccount(userAccount: string): Promise<Loan[]> {
+        const result: Loan[] = [];
+        for (const loanManager of this.getAllLoanManagers()) {
+            const lmLoans: Loan[] = await loanManager.getLoansForAccount(userAccount);
+            result.push(...lmLoans);
+        }
+        return result;
```

Comment:
use `Promise.all` to do requests in parallel, instead of sequentially `await`-ing.

Reply:
done